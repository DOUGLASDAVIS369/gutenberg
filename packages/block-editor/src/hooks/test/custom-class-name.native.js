/**
 * External dependencies
 */
import {
	addBlock,
	fireEvent,
	changeTextOfTextInput,
	getEditorHtml,
	initializeEditor,
	openBlockSettings,
	screen,
	setupCoreBlocks,
} from 'test/helpers';

setupCoreBlocks( [ 'core/paragraph' ] );

describe( 'custom class name', () => {
	it( 'should set the HTML class attribute on the block', async () => {
		// Arrange
		await initializeEditor();
		const block = await addBlock( screen, 'Paragraph' );
		await openBlockSettings( screen, block );

		// Act
		await fireEvent.press(
			screen.getByLabelText( 'Additional CSS class(es). Empty' )
		);
		await changeTextOfTextInput(
			await screen.getByPlaceholderText( 'Add classes' ),
			'my-class-name another-class-name'
		);

		// Assert
		expect( getEditorHtml() ).toMatchSnapshot();
	} );
} );
