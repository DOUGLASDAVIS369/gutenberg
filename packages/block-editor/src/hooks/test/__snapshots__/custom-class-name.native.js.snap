// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`custom class name should set the HTML class attribute on the block 1`] = `
"<!-- wp:paragraph {"className":"my-class-name another-class-name"} -->
<p class="my-class-name another-class-name"></p>
<!-- /wp:paragraph -->"
`;
